plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -130,7 +136,14 @@ public Page getNextPage()
     {
         Page page;
         try {
-            page = recordReader.nextPage();
+            if (remainingPage.isPresent()) {
+                page = remainingPage.get();
+                remainingPage = Optional.empty();
+                localMemoryContext.setBytes(0);
```

Comment:
why? document

Reply:
btw i think we can keep one page without declaring it's memory consumption.
same in the other class

Reply:
I would keep it. We account for page sizes when we process queries. I will add a comment.